src/Microsoft.ML.PipelineInference/Macros/PipelineSweeperMacro.cs

```diff
@@ -88,6 +112,77 @@ public static Output ExtractSweepResult(IHostEnvironment env, ResultInput input)
             return new Output { Results = outputView, State = autoMlState };
         }
 
+        private static RoleMappedData GetDataRoles(IHostEnvironment env, Arguments input)
+        {
+            var roles = new List<KeyValuePair<RoleMappedSchema.ColumnRole, string>>();
+
+            if (input.LabelColumns != null)
```

Comment:
>if (input.LabelColumns != null) [](start = 12, length = 31)

would you want to make specifying  this mandatory?

Reply:
one scenario that needs to be supported is  when user does not specify anything.  in that case case  we will do AutoInference on all columns.  

That's why these arguments (including Label) are not mandatory.   These arguments are meant to give user  the ability to override  the auto inference   (if needed)

---
In reply to: [202428925](https://github.com/dotnet/machinelearning/pull/461#discussion_r202428925) [](ancestors = 202428925)